components/snowflake/snowflake.app.js

```diff
@@ -0,0 +1,54 @@
+const { promisify } = require("util");
+const snowflake = require("snowflake-sdk");
+
+module.exports = {
+  app: "snowflake",
+  type: "app",
+  methods: {
+    async _getConnection() {
+      if (this.connection) {
+        return this.connection;
+      }
+
+      this.connection = snowflake.createConnection(this.$auth);
+      await promisify(this.connection.connect)();
```

Comment:
I'm sure you've tested this but kind of surprised connect doesn't use `this`... (as in no bind required)

Reply:
Hmm that's actually a good point. I looked at the code and everything indicates that this shouldn't work 😆 
I'll bind it to be safe.